python/common/org/python/types/Generator.java

```diff
@@ -1,13 +1,23 @@
 package org.python.types;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Possible states of a generator:
+ * 1. yield_point == 0    : The generator is 'fresh' i.e. (just started)
+ * 2. expression  == null : The generator is exhausted
+ * 3. message     != none : The generator currently holds value sent from caller via send() method
+ * 4. exception   != null : The generator currently holds exception sent from caller via throw() method
```

Comment:
Are any of these states mutually exclusive? e.g., can I have both a message *and* an exception?

Reply:
Yes every states are mutually exclusive